### DISCUSSION TITLE
Adding before saving/updating hook, or adding data as it was before in parameters

### DISCUSSION THREAD

**Discussion by nicooprat** [CodeDredd/pinia-orm#1762] [Ideas]
Hi there,

Would be handy to be able to compare previous and new data before saving. In my use case, I would like to merge a nested key content, because different APIs return only part of the data and I don't want to erase what I already got.

I mean something like the `watch` function does in Vue:

```ts
static saving(model, record, oldModel) {
  // Simplest merge
  model.foo = { ...oldModel.foo, ...model.foo }
}
```

Would it be possible?

I didn't find any way to do it for now, but I may have missed something.

Thanks!

**Comment by CodeDredd** [comment]
@nicooprat I patched the hooks. So now you can do what you like. The model is now the "updated" part and record is the old.

**Reply by nicooprat** [reply]
Thanks I'll try it! Is this part of the 1.9.0 release?

**Reply by CodeDredd** [reply]
Yeah. But i changed the design again back how it was. The only differnce is that now `$original` and `$isDirty` are working correctly. So by using `$original` you are getting the old model.
